src/AspNet.Security.OAuth.Salesforce/SalesforceAuthenticationOptions.cs

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace AspNet.Security.OAuth.Salesforce
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="SalesforceAuthenticationHandler"/>.
+    /// </summary>
+    public class SalesforceAuthenticationOptions : OAuthOptions
+    {
+        public SalesforceAuthenticationOptions OverrideHostInEndpoints(string newHost)
```

Comment:
It's pretty common in Salesforce land to think of and use test.salesforce.com in dev / test / stage environments and login.salesforce.com in production environments. At the same time, it seems unnecessary to have to change the endpoints to the full https://test.salesforce.com/services/oauth2/authorize and https://test.salesforce.com/services/oauth2/token for those non-prod environments, so this method makes it easier to simply override the hosts without having to look up and set the full endpoints when first getting this going.

That said, your lib, and so I'm happy to remove if you'd prefer.